src/Tempest/Http/src/RouteDiscovery.php

```diff
@@ -24,21 +28,31 @@ public function discover(ClassReflector $class): void
                 continue;
             }
 
-            $this->routeConfig->addRoute($method, $routeAttribute);
+            $routeAttribute->setHandler($method);
+
+            $this->routeConfigConstructor->addRoute($routeAttribute);
+        }
+    }
+
+    #[EventHandler(KernelEvent::BOOTED)]
+    public function finishDiscovery(): void
+    {
+        if ($this->routeConfigConstructor->hasRoutes()) {
```

Comment:
We do need to check if we have any new routes to apply because we have two paths coming here, when caching is not enabled it will be hit once by the KernelEvent::Booted. But with caching enabled it can be it two times in one run since we need the RouteConfig fully constructed when creating the cache as well, which is called before the booted event. Thus it will be called twice.